.editorconfig

```
@@ -0,0 +1,74 @@
+# Visual Studio generated .editorconfig file with C++ settings.
+root = true
+
+[*.{c++,cc,cpp,cppm,cxx,h,h++,hh,hpp,hxx,inl,ipp,ixx,tlh,tli}]
+
+indent_style = space
+indent_size = 4
+insert_final_newline = true
+charset = utf-8
```

What about adding _trim_trailing_whitespace_ from the [supported settings](https://learn.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options?view=vs-2022#supported-settings)?
```suggestion
trim_trailing_whitespace = true
charset = utf-8
```
As to the _end_of_line_, there are obviously some [issues](https://github.com/editorconfig/editorconfig/issues/226) with cross-platform/Git usage - so probably leave that one untouched.

---

Iirc, I left out trim_trailing_whitespace because VS's auto-format did not seem to apply this setting. So, we'd have a case where "code does not conform to settings" (or need to do this formatting change with some other tool). And - I have to confess, I don't see a problem with trailing whitespaces, or  - I can imaging cases where I'd want to have trailing whitespaces (e.g. in raw string literals) and I am not sure whether the editor is then smart enough to decide when it is fine to remove trailing whitespaces and when not. But I can not imagine a case where a trailing whitespace is really remotely troublesome. So - I am lacking experience here whether this is advantageous or not.
Hmm, should we then better write
`trim_trailing_whitespace = `
Is this better than just leaving it out?

---

regarding "end_of_line" - the line of reasoning here is something like: there haven't been problems with it so far, Git/IDEs/all other tooling seem to work fine, don't mess with it until there is a reason.

---

I agree. Feel free to skip the suggestion and leave as-is.

---

I removed the tab_width since (following https://editorconfig.org/):
tab_width: a whole number defining the number of columns used to represent a tab character. This **defaults to the value of indent_size and doesn't usually need to be specified**.